src/qt/bitcoinunits.cpp

```diff
@@ -150,6 +153,15 @@ QString BitcoinUnits::formatHtmlWithUnit(int unit, const CAmount& amount, bool p
     return QString("<span style='white-space: nowrap;'>%1</span>").arg(str);
 }
 
+QString BitcoinUnits::formatWithPrivacy(int unit, const CAmount& amount, SeparatorStyle separators, bool privacy)
+{
+    assert(amount >= 0);
+    QString value = format(unit, privacy ? 0 : amount, false, separators, true);
+    if (privacy) {
+        value.replace('0', '#');
+    }
+    return value + QString(" ") + shortName(unit);
+}
```

Comment:
micro-nit, can perform one less conditional (with slightly improved readability) by avoiding the embedded ternary:
```diff
-    QString value = format(unit, privacy ? 0 : amount, false, separators, true);
+    QString value;
     if (privacy) {
+        value = format(unit, 0, false, separators, true);
         value.replace('0', '#');
+    } else {
+        value = format(unit, amount, false, separators, true);
     }
```

Reply:
[Updated](https://github.com/bitcoin/bitcoin/pull/16432#issuecomment-633063508).